src/openfl/_internal/renderer/opengl/batcher/MultiTextureShader.hx

```diff
@@ -54,6 +70,50 @@ class MultiTextureShader {
 		gl.uniformMatrix4fv(uProjMatrix, 0, false, projectionMatrix);
 	}
 
+	static function compileShader(gl:GLRenderContext, source:String, type:Int):Null<GLShader> {
+		var shader = gl.createShader(type);
+		gl.shaderSource(shader, source);
+		gl.compileShader(shader);
+		
+		if (gl.getShaderParameter(shader, gl.COMPILE_STATUS) == 0) {
+			var message = gl.getShaderInfoLog(shader);
+			gl.deleteShader(shader);
+			Log.warn(message);
+			return null;			
+		}
+		
+		return shader;
+	}
+	
+	static function createProgram(gl:GLRenderContext, vertexSource:String, fragmentSource:String):Null<GLProgram> {
+		var vertexShader = compileShader(gl, vertexSource, gl.VERTEX_SHADER);
+		if (vertexShader == null) {
+			return null;
+		}
+
+		var fragmentShader = compileShader(gl, fragmentSource, gl.FRAGMENT_SHADER);
+		if (fragmentShader == null) {
+			if (vertexShader != null) gl.deleteShader(vertexShader);
```

Comment:
We always have a vertex shader here right? I would remove the check (also because you are not checking after linking again).